File: 45. Jump Game II/jump.py
class jump:
    def solution(self, nums: List[int]) -> int:
        if len(nums) < 1:
            return 0
        res = 0
        end = 0
        maxPos = 0
        for i in range(len(nums) - 1):
            maxPos = max(maxPos, nums[i] + i)
            if i == end:
                end = maxPos
                res = res + 1
        return res

File: 45. Jump Game II/jump.java
class jump {
    public int solution(int[] nums) {
        if (nums.length < 1) {
            return 0;
        }
        int res = 0, end = 0, maxPos = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            maxPos = Math.max(maxPos, i + nums[i]);
            if (i == end) {
                end = maxPos;
                res ++;
            }
        }
        return res;
    }
}